pkg/controller/namespace/namespace_controller_utils.go

```diff
@@ -459,11 +460,18 @@ func ServerPreferredNamespacedGroupVersionResources(discoveryClient discovery.Di
 	if err != nil {
 		return results, err
 	}
+
+	allErrs := []error{}
 	for _, apiGroup := range serverGroupList.Groups {
-		preferredVersion := apiGroup.PreferredVersion
```

Comment:
On second thought, perhaps we don't need the compensation here, because usually people upgrade the controller manager along with the apiserver. What do you think?